backend/elasticsearch-client/jdk-rest-client/src/main/java/org/hibernate/search/backend/elasticsearch/client/jdk/cfg/spi/RestJdkClient.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.search.backend.elasticsearch.client.jdk.cfg.spi;
+
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+import org.hibernate.search.util.common.annotation.Incubating;
+import org.hibernate.search.util.common.impl.Closer;
+
+@Incubating
+public class RestJdkClient implements AutoCloseable {
+
+	private final NodeProvider nodeProvider;
+
+	private HttpClient httpClient;
+
+	public RestJdkClient(NodeProvider nodeProvider, HttpClient httpClient) {
+		this.nodeProvider = nodeProvider;
+		this.httpClient = httpClient;
+	}
+
+	public NodeProvider.ServerNode nextNode() {
+		return nodeProvider.nextNode();
+	}
+
+	public <T> CompletableFuture<HttpResponse<T>> sendAsync(HttpRequest request,
+			HttpResponse.BodyHandler<T> responseBodyHandler) {
+		return httpClient.sendAsync( request, responseBodyHandler );
+	}
+
+	@Override
+	public void close() throws Exception {
+		if ( httpClient != null ) {
+			try ( Closer<Exception> closer = new Closer<>() ) {
+				Optional<Executor> executor = httpClient.executor();
+				// May look a bit silly ... but close was only added in JDK 21:
+				if ( ( (Object) httpClient ) instanceof AutoCloseable closeable ) {
+					closer.push( AutoCloseable::close, closeable );
+				}
+				if ( executor.isPresent() && executor.get() instanceof AutoCloseable closeable ) {
+					closer.push( AutoCloseable::close, closeable );
+				}
+			}
+			finally {
+				httpClient = null;
+			}
+		}
```

Comment:
Why does the closing look as it does ? well .....
Prior to JDK 21, this client does not expose a way to close it.... and it starts a thread and an executor (unless provided, which we do here), which are supposedly stopped and cleaned up once the client is not referenced from anywhere ... 
But TBH .. I'd rather limit the use of this client to JDK 21+ 🙂

Reply:
>  But TBH .. I'd rather limit the use of this client to JDK 21+ 🙂

+1 that's a perfectly reasonable limitation.